File: src/code-samples/langsmith/smithdb-migration/runs-retrieve-basic-before.kt
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 21
//KOTLIN 2.2.0
//DEPS com.langchain.smith:langsmith-java:0.1.0-beta.11

// :snippet-start: runs-retrieve-basic-before-kt
// :codegroup-tab: Before
import com.langchain.smith.client.LangsmithClient
import com.langchain.smith.client.okhttp.LangsmithOkHttpClient
// :remove-start:
import com.langchain.smith.models.runs.RunQueryParams
import com.langchain.smith.models.sessions.SessionListParams
// :remove-end:

// :remove-start:
fun main() {
    if (System.getenv("LANGSMITH_API_KEY").isNullOrBlank()) {
        println("[smithdb-runs-retrieve-basic-before] Skipping (LANGSMITH_API_KEY is not set).")
        return
    }

// :remove-end:
val client: LangsmithClient = LangsmithOkHttpClient.fromEnv()

var runId = "<run-id>"
// :remove-start:
val project = client.sessions().list(
    SessionListParams.builder().name("default").limit(1L).build()
).items().first()
val foundRun = client.runs().query(
    RunQueryParams.builder().addSession(project.id()).limit(1L).build()
).items().first()
runId = foundRun.id()
// :remove-end:
val run = client.runs().retrieve(runId)
println("${run.name()} ${run.status()} ${run.totalTokens()}")
// :remove-start:
}
// :remove-end:
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/runs-retrieve-basic-before.go
// :snippet-start: runs-retrieve-basic-before-go
// :codegroup-tab: Before
package main

import (
	"context"
	"fmt"

	"github.com/langchain-ai/langsmith-go"
)

// :remove-start:
func main() {
// :remove-end:
ctx := context.Background()
client := langsmith.NewClient()

runID := "<run-id>"
// :remove-start:
sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
	Name:  langsmith.F("default"),
	Limit: langsmith.F(int64(1)),
})
if err != nil {
	panic(err.Error())
}
found, err := client.Runs.Query(ctx, langsmith.RunQueryParams{
	Session: langsmith.F([]string{sessions.Items[0].ID}),
	Limit:   langsmith.F(int64(1)),
})
if err != nil {
	panic(err.Error())
}
runID = found.Runs[0].ID
// :remove-end:
run, err := client.Runs.Get(ctx, runID, langsmith.RunGetParams{})
// :remove-start:
if err != nil {
	panic(err.Error())
}
// :remove-end:
fmt.Println(run.Name, run.Status, run.TotalTokens)
// :remove-start:
}

// :remove-end:
// :snippet-end:
